apis/buttons/src/lib.rs

```diff
@@ -0,0 +1,140 @@
+#![no_std]
+
+use libtock_platform::{
+    share::Handle, subscribe::OneId, DefaultConfig, ErrorCode, Subscribe, Syscalls, Upcall,
+};
+
+/// The Buttons driver
+///
+/// # Example
+/// ```ignore
+/// use libtock2::Buttons;
+///
+/// // Read button state
+/// Buttons::is_pressed(0);
+///
+/// // Register for events
+///
+/// let listener = ButtonListener(|button, state| {
+///     // make use of the button's state
+/// });
+///
+/// share::scope(|subscribe| {
+///     if let Ok(()) = Buttons::register_listener(&listener, subscribe) {
+///         // yield
+///     }
+/// });
+/// ```
+pub struct Buttons<S: Syscalls>(S);
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum ButtonState {
+    Pressed,
+    Released,
+}
+
+pub fn convert(value: u32) -> ButtonState {
```

Comment:
`convert` isn't a very descriptive name for this function (I didn't think about the name while making size measurements). Using `From<u32>` makes sense to me, or you can name it something like `u32_to_button_state`.